Tests/HTTPRouteTests.swift

```diff
@@ -147,4 +147,112 @@ final class HTTPRouteTests: XCTestCase {
+                                              .init(name: "fish", value: "chips")])
+        )
+
+        XCTAssertFalse(
+            route ~= HTTPRequest.make(method: .GET,
+                                      path: "/mock",
+                                      query: [.init(name: "cat", value: "dog"),
+                                              .init(name: "fish", value: "squid")])
+        )
+    }
+
+    func testMultipleQueryItems_MatchesRoute() {
+        let route = HTTPRoute("GET /mock?fish=chips&cats=dogs")
+
+        XCTAssertTrue(
+            route ~= HTTPRequest.make(method: .GET,
+                                      path: "/mock",
+                                      query: [.init(name: "fish", value: "chips"),
+                                              .init(name: "cats", value: "dogs")])
+        )
+
+        XCTAssertTrue(
+            route ~= HTTPRequest.make(method: .GET,
+                                      path: "/mock",
+                                      query: [.init(name: "cats", value: "dogs"),
+                                              .init(name: "fish", value: "chips")])
+        )
+
+        XCTAssertFalse(
+            route ~= HTTPRequest.make(method: .GET,
+                                      path: "/mock",
+                                      query: [.init(name: "fish", value: "chips")])
+        )
+
+        XCTAssertFalse(
+            route ~= HTTPRequest.make(method: .GET,
+                                      path: "/mock",
+                                      query: [.init(name: "cats", value: "dogs")])
+        )
+
+        XCTAssertFalse(
+            route ~= HTTPRequest.make(method: .GET,
+                                      path: "/mock",
+                                      query: [.init(name: "cat", value: "dog"),
+                                              .init(name: "fish", value: "squid")])
+        )
+    }
+
+    func testQueryItemWildcard_MatchesRoute() {
+        let route = HTTPRoute("GET /mock?fish=*")
+
+        XCTAssertTrue(
+            route ~= HTTPRequest.make(method: .GET,
+                                      path: "/mock",
+                                      query: [.init(name: "fish", value: "chips")])
+        )
+
+        XCTAssertTrue(
+            route ~= HTTPRequest.make(method: .GET,
+                                      path: "/mock",
+                                      query: [.init(name: "cat", value: "dog"),
+                                              .init(name: "fish", value: "squid"),
+                                              .init(name: "fish", value: "chips")])
+        )
+
+        XCTAssertTrue(
+            route ~= HTTPRequest.make(method: .GET,
+                                      path: "/mock",
+                                      query: [.init(name: "cat", value: "dog"),
+                                              .init(name: "fish", value: "squid")])
+        )
```

Comment:
Minor: may be worth a negative test as well?

```suggestion
        )
        
         XCTAssertFalse(
             route ~= HTTPRequest.make(method: .GET,
                                       path: "/mock",
                                       query: [.init(name: "cat", value: "dog")])
         )
```